x/concentrated-liquidity/position_test.go

```diff
@@ -179,7 +210,113 @@ func (s *KeeperTestSuite) TestGetPosition() {
 				s.Require().NoError(err)
 				s.Require().Equal(test.expectedPosition, position)
 			}
-
 		})
 	}
 }
+
+// Thorough fuzz testing for join-time sumtrees
+func (s *KeeperTestSuite) TestMultiplePositionsFuzz() {
+	// Init suite for fuzz testing
+	s.Setup()
+
+	// Create a default CL pool
+	s.PrepareConcentratedPool()
+
+	// Test 100 random positions in same pool at random jointimes
+	// Note: we can fuzz these values if needed as well
+	numPositions := 100
+	validPoolId := uint64(1)
+
+	// Max lower and upper ticks
+	tickLowerBound := int64(100)
+	tickUpperBound := int64(100)
+
+	// Max time elapsed between positions (in seconds)
+	timeElapsedUpperBound := 1000
+
+	// We generate a new test account per position for robustness
+	testAccounts := osmoutils.CreateRandomAccounts(numPositions)
```

Comment:
I feel like, for robustness, we should have accounts with multiple positions (this feels like the more complex logic imo)